### DISCUSSION TITLE
Disabling specific cells

### DISCUSSION THREAD

**Discussion by InjAnnuity1** [anvilistas/tabulator#99] [Q&A]
In my app, it makes sense to disable some cells, based on information from the row and column.

For example, suppose a column represents an insurance policy, issued on a particular date, and a row represents one or more dated values.  That is, the "effective date" is part of the row.  Thus, we have two dates for each cell: an issue date, and an effective date.

If the effective date comes _before_ the issue date, then I would like to disable the cell.  After all, the value can't apply to a policy that doesn't exist yet, at that given date.  

Some visual feedback for the user would be nice, too.  E.g., the cell should display a "n/a", or the mouse cursor should change to 🚫 when over a non-editable cell.

This logic applies to _most_ of the editable cells in the table.  The "effective date" column must still be editable, though.

Working from the Tabulator documentation, it's far from clear how to do any of the above.  Any pointers would be appreciated.

**Comment by lewysigns** [comment]
I am not sure how to change the cursor but you can import the [popup module](https://tabulator.info/docs/5.4/menu#popup-programmatic) detail on how to add modules [here](https://github.com/s-cork/Tabulator#modules).

Then on a cell_click event you can check the values you are interested in to determine if the popup appears. Example [here](https://github.com/s-cork/Tabulator/discussions/100#discussioncomment-5397229)

**Reply by InjAnnuity1** [reply]
Thanks.  

If the user has already clicked on a disabled cell, then I haven't done my job.  I'm trying to let the user know _in advance_ that clicking (or otherwise activating) the cell would be useless, so that they don't even bother trying.

But if they click on it anyway, then nothing whatsoever should happen.  No response.  No editor should appear, for example.  That cell is intended to be not active in any way.

**Reply by lewysigns** [reply]
I see.

So you'll want to use the [edit module](https://tabulator.info/docs/5.4/edit#overview)  where you can have a function check if a field is editable when the table is being rendered using the [editable ](https://tabulator.info/docs/5.4/edit#optional) keyword.

You can have a [tooltip](https://tabulator.info/docs/5.4/menu#tooltips-cell) to alert the user, but still haven't done an icon.

**Reply by InjAnnuity1** [reply]
Yes!  Thank you!  I was searching Tabulator's docs for "enabled" and "disabled", and didn't think to search for "editable".

**Reply by lewysigns** [reply]
Yeah nomenclature is funny when you're jumping between projects haha.
